Drift/Private/DriftFlexmatch.cpp

```diff
@@ -50,41 +56,72 @@ TStatId FDriftFlexmatch::GetStatId() const
 
 void FDriftFlexmatch::ReportLatencies()
 {
+	// Accumulate a mapping of regions->ping and once all results are in, PATCH drift-flexmatch
+	TSharedRef<TMap<FString, int>, ESPMode::ThreadSafe> LatenciesByRegion(new TMap<FString, int>());
 	const auto HttpModule = &FHttpModule::Get();
 	for(auto Region: PingRegions)
 	{
 		const auto Request = HttpModule->CreateRequest();
 		Request->SetVerb("GET");
 		Request->SetURL(FString::Format(*PingUrlTemplate, {Region}));
 		Request->OnProcessRequestComplete().BindLambda(
-		[this, Region](FHttpRequestPtr Request, FHttpResponsePtr Response, bool bConnectedSuccessfully)
+		[this, Region, LatenciesByRegion](FHttpRequestPtr Request, FHttpResponsePtr Response, bool bConnectedSuccessfully)
 		{
 			if (!bConnectedSuccessfully)
 			{
 				UE_LOG(LogDriftMatchmaking, Error, TEXT("FDriftFlexmatch::ReportLatencies - Failed to connect to '%s'"), *Request->GetURL());
-				return;
+				LatenciesByRegion->Add(Region, -1);
 			}
-			if ( ! (DoPings && RequestManager) )
+			else
 			{
-				return;
+				LatenciesByRegion->Add(Region, static_cast<int>(Request->GetElapsedTime() * 1000));
 			}
-			JsonValue Payload{rapidjson::kObjectType};
-			JsonArchive::AddMember(Payload, TEXT("region"), *Region);
-			JsonArchive::AddMember(Payload, TEXT("latency_ms"), Request->GetElapsedTime() * 1000);
-			auto PatchRequest = RequestManager->Patch(FlexmatchURL, Payload, HttpStatusCodes::Ok);
-			PatchRequest->OnError.BindLambda([this](ResponseContext& context)
+			// if all regions have been added to the map, report back to drift ...
+			if (LatenciesByRegion->Num() == PingRegions.Num())
```

Comment:
This lambda becomes rather complex here, and both accumulates all the responses, and sends them on to Drift. Perhaps break the sending part out to a new member function?